sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java

```diff
@@ -392,17 +392,15 @@ public static String toTableSpec(TableReference ref) {
 
   /**
    * Parse a table specification in the form {@code "[project_id]:[dataset_id].[table_id]"} or
-   * {@code "[dataset_id].[table_id]"}.
+   * {@code "[project_id].[dataset_id].[table_id]"} or {@code "[dataset_id].[table_id]"}.
    *
    * <p>If the project id is omitted, the default project id is used.
    */
   public static TableReference parseTableSpec(String tableSpec) {
     Matcher match = BigQueryIO.TABLE_SPEC.matcher(tableSpec);
     if (!match.matches()) {
       throw new IllegalArgumentException(
-          "Table reference is not in [project_id]:[dataset_id].[table_id] "
-              + "format: "
-              + tableSpec);
+          "Table reference is not in the expected " + "format: " + tableSpec);
```

Comment:
can you add the expected formats in the error message? so that users will receive actionable errors

Reply:
Sure. Done. @pabloem